### PR TITLE
Add support for generic parameters in make_set_digest function

### PR DESCRIPTION
**Pull request by Akanksha-kedia**
MicrosoftTeams-image (17)
MicrosoftTeams-image (18)

### CODE REVIEW

**Comment by tdcmeehan** [comment]
Can you add some unit tests for this change?  I am not sure of the consequences--one unit test for every major type?  Thanks!

**Comment by tdcmeehan** [comment]
Thank you for adding these tests.  To be safe, let's make them round-trip, i.e. take the output binary, re-serialize it into a set digest, and check to see if the elements are present.

**Comment by tdcmeehan** [comment]
`TestTDigestAggregationFunction` has examples of round-tripping.  Basically, you need to take that output binary, then construct a SetDigest from it and infer information about the structure that is consistent with the inputs of the test.

**Comment by Akanksha-kedia** [comment]
@tdcmeehan kindly review and help merge.

**Comment by tdcmeehan** [comment]
This is better, but what I had in mind was this:

1. Create a set digest from values, then SELECT back the varbinary as hex.
2. Convert the hex to a format understandable and castable to varbinary, then select values back from the set digest.


TDigest tests above do this as an example.

**Comment by Akanksha-kedia** [comment]
I actually tried lot to do this.

**Comment by Akanksha-kedia** [comment]
@tdcmeehan please check the changes and merge, getting values from setdigest i m not able to actually achieve that.As a new requirement we can take up this

File: presto-main/src/test/java/com/facebook/presto/type/setdigest/TestMakeSetDigest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        assert "0.0".equals(assertions.getQueryRunner().execute(
+                "SELECT jaccard_index(make_set_digest(value), make_set_digest(value1)) FROM (VALUES ('a', 'b'),('c', 'd')) T(value,value1)").getMaterializedRows().get(0).getField(0).toString());
+
+        String hashCount = "{" + Murmur3Hash128.hash64(Slices.utf8Slice("c")) + "=1}";
+    }
+
+    @Test
+    public void testMakeSetDigestVarchar()
+    {
+        assert assertions.getQueryRunner().execute(
```
Comment (tdcmeehan) [inline presto-main/src/test/java/com/facebook/presto/type/setdigest/TestMakeSetDigest.java:106]: Please use TestNG assertions here and everywhere else in this file.  Do not use Java assert.

**Review by tdcmeehan — CHANGES_REQUESTED** [review]
In lieue of round tripping, let's not assert on binary contents.  If it ever changes, the test will have to be updated in a way that will be very hard to discover if there was a regression.  Instead, let's inspect the digests--cardinality, jaccard index and insersection.

**Comment by tdcmeehan** [comment]
Also please make sure the commit adheres to our guidelines (see CONTRIBUTING.md).  The title is too long.

**Comment by Akanksha-kedia** [comment]
Ok, will do that.

**Comment by tdcmeehan** [comment]
Screenshot 2023-09-22 at 10 47 10 AM
Looks like commit message still too wide.

**Comment by Akanksha-kedia** [comment]
i have shortened @tdcmeehan

**Review by tdcmeehan — APPROVED** [review]

**Comment by Akanksha-kedia** [comment]
@tdcmeehan please help to merge this PR.Thanks for review and approval !!!

**Pull request merged**